### PR TITLE
Blog for Lucky 0.30

### PR DESCRIPTION
**Pull request by jwoertink**

### CODE REVIEW

File: src/posts/lucky_0.30.cr [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+  title "Small updates with a huge step forward"
+  slug "lucky-0_30-release"
+
+  def published_at : Time
+    Time.utc(year: 2022, month: 4, day: 11)
+  end
+
+  def summary : String
+    <<-TEXT
+    This new release of lucky 0.30 is one of our smaller releases, but
```
Comment (edwardloveall) [inline src/posts/lucky_0.30.cr]:
```suggestion
    Lucky 0.30 is one of our smaller releases, but it
```

File: src/posts/lucky_0.30.cr [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+    takes a huge step forward with the inclusion of automated security testing.
+    TEXT
+  end
+
+  def content : String
+    <<-MD
+    Lucky 0.30 is a smaller release compared to previous ones, but this one
```
Comment (edwardloveall) [inline src/posts/lucky_0.30.cr]:
```suggestion
    Lucky 0.30 is a smaller release compared to previous ones, but it
```

File: src/posts/lucky_0.30.cr [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+    includes some pretty big steps forward. Better error messages, some spring
```
Comment (edwardloveall) [inline src/posts/lucky_0.30.cr]:
```suggestion
    includes some big steps forward. Better error messages, some spring
```

File: src/posts/lucky_0.30.cr [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+    cleaning, a few added methods, and the new inclusion of automated security
+    testing that we're super excited about!
+
+    We will cover a few notable changes, but you can view all the changes
```
Comment (edwardloveall) [inline src/posts/lucky_0.30.cr]:
```suggestion
    We will cover a few notable differences, but you can view all the changes
```

File: src/posts/lucky_0.30.cr [hunk 5]
```diff
@@ -0,0 +1,167 @@
...
+    from the [CHANGELOG](https://github.com/luckyframework/lucky/blob/main/CHANGELOG.md)
+
+    ### How to upgrade
+
+    See the [UPGRADE_NOTES](https://github.com/luckyframework/lucky/blob/main/UPGRADE_NOTES.md#upgrading-from-029-to-030).
+
+    Or you can view the changes using [LuckyDiff](https://luckydiff.com/?from=0.29.0&to=0.30.0).
```
Comment (edwardloveall) [inline src/posts/lucky_0.30.cr]:
```suggestion
    You can also view the changes using [LuckyDiff](https://luckydiff.com/?from=0.29.0&to=0.30.0).
```

File: src/posts/lucky_0.30.cr [hunk 6]
```diff
@@ -0,0 +1,167 @@
...
+    This tool helps see what has changed between versions and will be included in upgrade notes from now on.
+
+    ## Here's what's new
+
+    ### Bright Security (formerly Neuralegion)
+
+    During this release, we've worked directly with the [Bright Security](https://brightsec.com/)
+    team to bring out-of-the-box security testing support for Lucky apps!
+
+    When we say "security testing", what we mean is catching vulnerabilities such as
```
Comment (edwardloveall) [inline src/posts/lucky_0.30.cr]:
```suggestion
    When we say "security testing", what we mean is catching vulnerabilities such as:
```

File: src/posts/lucky_0.30.cr [hunk 7]
```diff
@@ -0,0 +1,167 @@
...
+    We've decided to just ditch the magic `--` character, and go a little more generic with a
+    Base64 JSON string. The old way will now be deprecated, and removed in the next release.
+
+    [Read more on this PR](https://github.com/luckyframework/lucky/pull/1674)
+
+    ### Remote IPs from the client
+
+    It's pretty common that you may need to get the client's IP. Maybe you want to track
+    where they're logging in from, or maybe you want to do a little geocoding on their
+    location? Whatever the reason, it's a nice thing to be able to easily access.
```
Comment (edwardloveall) [inline src/posts/lucky_0.30.cr]:
```suggestion
    It's pretty common that you may need to get the client's IP. Maybe you need to log it for analytics,
    or maybe you want to do a little geocoding on their
    location. Whatever the reason, it's a nice thing to be able to easily access.
```

File: src/posts/lucky_0.30.cr [hunk 8]
```diff
@@ -0,0 +1,167 @@
...
+    Lucky::RemoteIpHandler.configure do |settings|
+      # optionally use the X-Real-IP header instead
+      settings.ip_header_name = "X-Real-Ip"
+    end
+    ```
+
+    ### Slugify Integration
+
+    When you want to use slugs for your URLs in place of passing an ID, you can turn
+    to Avram to "slugify" a value for you. Previously we had the `AvramSlufigy` shard
```
Comment (edwardloveall) [inline src/posts/lucky_0.30.cr]:
```suggestion
    to Avram to "slugify" a value for you. Previously we had the `AvramSlugify` shard
```

File: src/posts/lucky_0.30.cr [hunk 9]
```diff
@@ -0,0 +1,167 @@
...
+    Please report any issues, and as always, PRs are greatly appreciated!
+
+    ### Follow and spread the word
+
+    If you haven't already, give us a [star on GitHub](https://github.com/luckyframework/lucky),
+    and be sure to follow us on [Twitter](https://twitter.com/luckyframework/).
+
+    Learn tips and tricks with [LuckyCasts](https://luckycasts.com/).
+
+    If you have any questions, or just want to chat,  please join us on [Discord](#{Chat::Index.path}).
```
Comment (edwardloveall) [inline src/posts/lucky_0.30.cr]:
```suggestion
    If you have any questions, or just want to chat, please join us on [Discord](#{Chat::Index.path}).
```

**Review by edwardloveall — APPROVED** [review]
🎉

**Comment by jwoertink** [comment]
Awesome! Thanks for the review @edwardloveall

**Review by matthewmcgarvey — COMMENTED** [review]
Personally, I put in a lot of effort in this release. Avram now has the ability to run specs in a transaction and LuckyFlow has an in-memory alternative to using Chrome. Both of those combined speeds up specs by 4-10x. Nox as well

**Comment by edwardloveall** [comment]
That's huge! Those both seem worth noting here.

**Comment by jwoertink** [comment]
I can add the stuff about the specs, but Flow wasn't updated in this release because I couldn't get an app to compile with it. I don't think LuckyCLI was ever updated with the latest LuckyFlow stuff.

File: src/posts/lucky_0.30.cr [hunk 10]
```diff
@@ -0,0 +1,187 @@
...
+    Following the Lucky mantra of catching bugs in development, it is our hope that this new integration
+    gives you peace of mind that your production app is a lot safer and solid.
+
+    ### Faster specs
+
+    After quite a bit of refactoring, we now have our database related specs running in a transaction
+    as opposed to truncating between specs. This has shown to give at least 4x speed to your specs.
+
+    ```crystal
+    # notes on how to use this
```
Comment (jwoertink) [inline src/posts/lucky_0.30.cr]: @matthewmcgarvey not sure if I'm missing something, but it doesn't look like LuckyCLI generates an app with the transactional spec helper stuff. Is it an opt-in feature?

File: src/posts/lucky_0.30.cr [hunk 11]
```diff
@@ -0,0 +1,215 @@
...
+    [For LuckyFlow](https://github.com/luckyframework/lucky_flow/pull/137)
+
+    ### Built-in Process runner Nox
+
+    [Nox](https://github.com/matthewmcgarvey/nox) is a process runner built by [@matthewmcgarvey](https://github.com/matthewmcgarvey).
+    Lucky now ships with this out of the box so there's no longer a need to install an additional tool for local development. After
+    installing LuckyCLI, you can boot your app locally using `lucky dev`, and this will start the Nox process manager using
+    your `Procfile.dev`.
+
+    You can continue to use other process managers such as Overmind, Foreman, or the Heroku CLI if you wish.
```
Comment (jwoertink) [inline src/posts/lucky_0.30.cr]: @edwardloveall this is the section that was added in.

File: src/posts/lucky_0.30.cr [hunk 12]
```diff
@@ -0,0 +1,215 @@
...
+    ```crystal
+    class ApiClient < Lucky::BaseHTTPClient
+      # Add this line
+      app AppServer.new
+
+      # ...
+    end
+    ```
+
+    Finally, if you want to try out even more speed for your Flow specs, you can upgrade to v0.8.0.
```
Comment (edwardloveall) [inline src/posts/lucky_0.30.cr]:
```suggestion
    Finally, if you want to try out even more speed for your Flow specs, you can upgrade to LuckyFlow v0.8.0.
```

File: src/posts/lucky_0.30.cr [hunk 13]
```diff
@@ -0,0 +1,215 @@
...
+    This release of LuckyFlow wasn't included for this 0.30.0 release of Lucky, but you can still include
+    it to test it out.
+
+    LuckyFlow 0.8.0 brings in a new Driver registration along with a new "Webless" driver. By using this
+    driver, no Chrome process is booted which saves on both CPU and memory. The caveat to this is the
+    limitations to any javascript related interactions.
```
Comment (edwardloveall) [inline src/posts/lucky_0.30.cr]:
```suggestion
    driver, no Chrome process is booted which saves on both CPU and memory. The caveat to this is that
    JavaScript is not enabled so those interactions will not function.
```

Is this correct? I wasn't sure what you meant by "the limitations to any javascript related interactions"

Comment (jwoertink) [reply src/posts/lucky_0.30.cr]: Exactly.

**Comment by jwoertink** [comment]
I'm gonna go ahead and merge this so we can get it out there. Thanks for the reviews!

**Pull request merged**